nil/services/journald_forwarder/clickhouse.go

```diff
@@ -0,0 +1,161 @@
+package journald_forwarder
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/ClickHouse/clickhouse-go/v2"
+	"github.com/ClickHouse/clickhouse-go/v2/lib/driver"
+	"github.com/NilFoundation/nil/nil/common/logging"
+)
+
+const (
+	DefaultDatabase = "nil"
+	DefaultTable    = "events"
+)
+
+type ClickhouseConfig struct {
+	ClickhouseAddr string
+	ListenAddr     string
+	DbUser         string
+	DbDatabase     string
+	DbPassword     string
+}
+
+type Clickhouse struct {
+	ClickhouseConfig
+	connect clickhouse.Conn
+}
+
+func (s *Clickhouse) InsertColumnsInTable(ctx context.Context,
+	database, tableName string,
+	columns, columnsType []string,
+) error {
+	columnClauses := make([]string, len(columns))
+	for i, column := range columns {
+		columnClauses[i] = fmt.Sprintf("ADD COLUMN %s %s", column, columnsType[i])
+	}
+
+	query := fmt.Sprintf(
+		"ALTER TABLE %s.%s %s;",
+		database,
+		tableName,
+		strings.Join(columnClauses, ", "),
+	)
+
+	return s.connect.Exec(ctx, query)
+}
+
+func (s *Clickhouse) InsertData(
+	ctx context.Context,
+	database string,
+	tableName string,
+	columns []string,
+	values [][]any,
+) error {
+	placeholders := make([]string, len(columns))
+	for i := range columns {
+		placeholders[i] = "?"
+	}
+
+	// Create multiple value placeholders for batch insert
+	valueRows := make([]string, len(values))
+	for i := range values {
+		valueRows[i] = fmt.Sprintf("(%s)", strings.Join(placeholders, ", "))
+	}
+
+	query := fmt.Sprintf(
+		"INSERT INTO %s.%s (%s) VALUES %s",
+		database,
+		tableName,
+		strings.Join(columns, ", "),
+		strings.Join(valueRows, ", "),
```

Comment:
Is there not some achievable limit on the length of the query that we might encounter?

Reply:
Only on query size as I know.

Reply:
Should we provide for handling such a situation (and reducing the size of the batch) or do we consider such a situation impossible in our installation?